UnleashClient/constraints/Constraint.py

```diff
@@ -1,17 +1,150 @@
-# pylint: disable=invalid-name, too-few-public-methods
+# pylint: disable=invalid-name, too-few-public-methods, use-a-generator
+from typing import Optional, Union
+from datetime import datetime
+from enum import Enum
+from dateutil.parser import parse, ParserError
+import semver
 from UnleashClient.utils import LOGGER, get_identifier
 
 
+class ConstraintOperators(Enum):
+    # Logical operators
+    IN = "IN"
+    NOT_IN = "NOT_IN"
+
+    # String operators
+    STR_ENDS_WITH = "STR_ENDS_WITH"
+    STR_STARTS_WITH = "STR_STARTS_WITH"
+    STR_CONTAINS = "STR_CONTAINS"
+
+    # Numeric oeprators
+    NUM_EQ = "NUM_EQ"
+    NUM_GT = "NUM_GT"
+    NUM_GTE = "NUM_GTE"
+    NUM_LT = "NUM_LT"
+    NUM_LTE = "NUM_LTE"
+
+    # Date operators
+    DATE_AFTER = "DATE_AFTER"
+    DATE_BEFORE = "DATE_BEFORE"
+
+    # Semver operators
+    SEMVER_EQ = "SEMVER_EQ"
+    SEMVER_GT = "SEMVER_GT"
+    SEMVER_LT = "SEMVER_LT"
+
+
 class Constraint:
     def __init__(self, constraint_dict: dict) -> None:
         """
         Represents a constraint on a strategy
 
         :param constraint_dict: From the strategy document.
         """
-        self.context_name = constraint_dict['contextName']
-        self.operator = constraint_dict['operator']
-        self.values = constraint_dict['values']
+        self.context_name: str = constraint_dict['contextName']
+        self.operator: ConstraintOperators = ConstraintOperators(constraint_dict['operator'].upper())
+        self.values = constraint_dict['values'] if 'values' in constraint_dict.keys() else []
+        self.value = constraint_dict['value'] if 'value' in constraint_dict.keys() else []
```

Comment:
else `[]'  seems a bit weird for a string?

Reply:
Maybe `None` is better?

Reply:
Whoops, that's an embarassing C/P error.  Def. should be None!